### PR TITLE
fix : #216 Mark as complete button UI sync 

### PR DESCRIPTION
**Pull request by Maawan**
Hi @hkirat , to solve this issue I observed that whenever a user is changing content then the sidebar reload data from backend and the checkbox is checked according to completed or not but the problem is only when a user is on the same page and tried to check or click on markascompleted then it could not see the changes in the sidebar and the main content screen accordingly. If sidebar again reloads the data then it will show the correct checked status.

My Approach
I created a atom for this, if any user is changing checkbox then this atom will hold its value and will reflect on the right side and vice versa. As user changes content page, sidebar again gets data from backend and the role of atom is complete.

I am just a beginner, please suggest for the improvement required  

I have attached the video after solving the issue 

https://github.com/code100x/cms/assets/40340661/bcf9c915-ad33-4848-bb55-dbd9991607e1

### CODE REVIEW

File: src/components/Sidebar.tsx [hunk 1]
```diff
@@ -58,9 +59,19 @@ export function Sidebar({
     const pathArray = findPathToContent(fullCourseContent, contentId);
     if (pathArray) {
       const path = `/courses/${courseId}/${pathArray.join('/')}`;
+      console.log(`Path is this ${path}`);
```
Comment (devsargam) [inline src/components/Sidebar.tsx]: could you remove this log

Comment (Maawan) [reply src/components/Sidebar.tsx]: In the commit named removed debug statements I have removed these logs

File: src/store/atoms/markAsComplete.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import {atom} from 'recoil';
+
+interface markAsCompleteParams {
+    isValid : boolean
+    path? : string,
+    isCompleted? : boolean
+}
+
+export const markAsCompleteAtom = atom<markAsCompleteParams>({
```
Comment (hkirat) [inline src/store/atoms/markAsComplete.ts:9]: This approach seems slightly ad-hoc-ish
You're using a single atom for all checkboxes
You probably need an atom family here

**Review by hkirat — CHANGES_REQUESTED** [review]

**Comment by Maawan** [comment]
@hkirat So you mean for every module of the course there should be an atom in the atom-family, atom's value should change on triggering checkbox or markasComplete button ?

**Comment by hkirat** [comment]
yes! each atom in the atomfamily should be identified using the courseId

**Comment by Maawan** [comment]
@hkirat Got your point, implementing it

**Comment by Maawan** [comment]
@hkirat This time I changed the approach and tried using Family .
My approach is, I am creating an atomFamily with the atoms of different different courses. 
And every atom will contains an array of objects in this structure 
[{ id : 2,
   parentId : 1,
   isCompleted : true},
{id : 3,
   parentId : 1,
   isCompleted : true}

]
Here id is the content id and parentId is the folder id which is helpfull to display the percentage completed in the FolderView component.

Whenever from Sidebar or ContentRenderer someone hits marksAsComplete or checks the box, firstly I will make the backend request and then update the state for that specific content id

Is this time my approach fine ?

I am stuck in a situation, in atomFamily using atomSelectors , nextJS is allowing me to use server function in client component. I need to getFullCourseContent function to get the course data and convert into this format [{ id : 2,
   parentId : 1,
   isCompleted : true},
{id : 3,
   parentId : 1,
   isCompleted : true}

] and store in state but next is not letting me allow it 

Any suggestions ?

**Comment by hkirat** [comment]
We need the data to be hydrated from the server for SEO

**Comment by hkirat** [comment]
yeah you cant set state on the server
read https://nextjs.org/docs/pages/building-your-application/data-fetching/get-server-side-props

**Pull request closed without merging**